### PR TITLE
fix the dicom finish gadget and client

### PR DESCRIPTION
**Pull request by gtdeveloper5**
fix the dicom finish gadget and fix the client for blob writer

### CODE REVIEW

File: gadgets/dicom/DicomFinishGadget.cpp [hunk 1]
```diff
@@ -436,10 +449,6 @@ namespace Gadgetron {
             WRITE_DCM_STRING(key, "3.0");
         }
 
-        // Spacing Between Slices
```
Comment (naegelejd) [inline gadgets/dicom/DicomFinishGadget.cpp]: Can you put this one back in?

**Comment by naegelejd** [comment]
Hui, thanks this looks good. Could you add the "Spacing Between Slices" back in?
```diff
-        // Spacing Between Slices
-        key.set(0x0018, 0x0088);
-        ACE_OS::snprintf(buf, BUFSIZE, "%f", r_space.fieldOfView_mm.z);
-        WRITE_DCM_STRING(key, buf);
```

**Comment by gtdeveloper5** [comment]
Any reason the patientBirthdate is not included in dicom?

        // Patient Birthdate
        key.set(0x0010, 0x0030);
        if (patient_info.patientBirthdate) {
            //ACE_OS::snprintf(buf, BUFSIZE, "%04d%02d%02d", patient_info.patientBirthdate().get().year(),
            //        patient_info.patientBirthdate().get().month(), patient_info.patientBirthdate().get().day());
            //WRITE_DCM_STRING(key, buf);
        }
        else {
            status = dataset->insertEmptyElement(key);
        }

**Comment by naegelejd** [comment]
No I don't know why that's commented. Could you uncomment that as well? Thanks

**Comment by naegelejd** [comment]
Almost, can you "erase" the '-' from the patient birthdate?
```c++
b.erase(std::remove(d.begin(), d.end(), '-'), d.end());
WRITE_DCM_STRING(key, b.c_str());
```

**Pull request merged**